driver/src/main/scala/api/MongoDriver.scala

```diff
@@ -136,6 +136,20 @@ class MongoDriver(
    * @param options $optionsParam
    */
   def connection(nodes: Seq[String], options: MongoConnectionOptions = MongoConnectionOptions(), authentications: Seq[Authenticate] = Seq.empty, name: Option[String] = None): MongoConnection = {
+    Await.result(askConnection(nodes, options, authentications, name), Duration.Inf)
+  }
+
+  /**
+   * Creates a new MongoConnection.
+   *
+   * $seeConnectDBTutorial
+   *
+   * @param nodes $nodesParam
+   * @param authentications $authParam
+   * @param name $connectionNameParam
+   * @param options $optionsParam
+   */
+  def askConnection(nodes: Seq[String], options: MongoConnectionOptions = MongoConnectionOptions(), authentications: Seq[Authenticate] = Seq.empty, name: Option[String] = None): Future[MongoConnection] = {
```

Comment:
I'm not 100% by the `askConnection` naming. It depends whether the former connection would be deprecated in favor of such one, what is not yet decided.

Reply:
I think I fixed all other review points. Let me known when this point is decided.

Reply:
https://github.com/cchantep/ReactiveMongo/commit/2c62e4ec4cd69929a2f887012865bb5478c43ebc

Reply:
👍

Reply:
You can cherry pick the API changes so this PR can go on